frame/contracts/src/exec.rs

```diff
@@ -75,6 +75,45 @@ where
 	}
 }
 
+/// combined key type for both fixed and variable storage keys.
+pub enum Key<T: Config> {
+	/// variant for fixed sized keys.
+	Fix(FixSizedKey),
+	/// variant for variable sized keys.
+	Var(VarSizedKey<T>),
+}
+
+impl<T: Config> Key<T> {
+	/// copies self into a new vec.
+	pub fn to_vec(&self) -> Vec<u8> {
+		match self {
+			Key::Fix(v) => v.to_vec(),
+			Key::Var(v) => v.to_vec(),
+		}
+	}
+}
+
+impl<T: Config> From<FixSizedKey> for Key<T> {
+	fn from(v: FixSizedKey) -> Self {
+		Key::Fix(v)
+	}
+}
+
+impl<T: Config> From<VarSizedKey<T>> for Key<T> {
+	fn from(v: VarSizedKey<T>) -> Self {
+		Key::Var(v)
+	}
+}
+
+impl<T: Config> StorageKey<T> for Key<T> {
+	fn hash(&self) -> Vec<u8> {
```

Comment:
I think we can get rid of the `StorageKey` trait and the type aliases now. We can just implement `hash` directly on the enum. And we don't need the aliases as we no longer refer to them on their own.

Reply:
Removed the trait, but the type aliases seems useful since we use it for a lot of into conversion.
e.g: https://github.com/paritytech/substrate/blob/a449aa66cd52f7870b9d004b290e4becdad6a6c9/frame/contracts/src/benchmarking/mod.rs#L1133-L1138

Reply:
Can't we just implement `fn from_var` and `fn from_fixed` on `Key` directly? We are just creating the inner type and then calling `into()` on it. This seems like an unnecessary extra step. Also, I try to avoid type aliases where possible :)